scripts/build-setup.sh

```diff
@@ -148,10 +148,39 @@ if run_step "1"; then
 
     # use conda-lock to create env
     conda-lock install --conda $(which conda) -p $CYDIR/.conda-env $LOCKFILE &&
-
     source $CYDIR/.conda-env/etc/profile.d/conda.sh &&
     conda activate $CYDIR/.conda-env
     exit_if_last_command_failed
+
+    # install circt into conda
+    git submodule update --init $CYDIR/conda-reqs/install-circt &&
+    $CYDIR/conda-reqs/install-circt/bin/download-release-or-nightly-circt.sh \
+        -f circt-full-shared-linux-x64.tar.gz \
+        -i $CONDA_PREFIX \
+        -v version-file \
+        -x $CYDIR/conda-reqs/circt.json \
+        -g null
+    exit_if_last_command_failed
+
+    # Conda Setup
+    # Provide a sourceable snippet that can be used in subshells that may not have
+    # inhereted conda functions that would be brought in under a login shell that
+    # has run conda init (e.g., VSCode, CI)
+    read -r -d '\0' CONDA_ACTIVATE_PREAMBLE <<'END_CONDA_ACTIVATE'
+if ! type conda >& /dev/null; then
+    echo "::ERROR:: you must have conda in your environment first"
+    return 1  # don't want to exit here because this file is sourced
+fi
+
+# if we're sourcing this in a sub process that has conda in the PATH but not as a function, init it again
+conda activate --help >& /dev/null || source $(conda info --base)/etc/profile.d/conda.sh
+\0
+END_CONDA_ACTIVATE
+
+    replace_content env.sh build-setup-conda "# line auto-generated by $0
```

Comment:
Spacing bug?

Reply:
There shouldn't be (assuming you are referring to `$CONDA_ACTIVATE_PREAMBLE` being at the start of the line)